mathesar/reflection.py

```diff
@@ -93,9 +92,6 @@ def reflect_new_table_constraints(table):
 def reflect_db_objects():
     if not cache.get(DB_REFLECTION_KEY):
         reflect_databases()
-        for database_key in get_non_default_database_keys():
```

Comment:
This code is essential. You can replace it by iterating through the `models.Database` objects instead.

Reply:
@kgodey I tried models.Database objects it was not iterating on them facing issue while iterating. Could you please suggest which one to iterate over to get them.

Reply:
@arvind-1 You should replace it with `for database in models.Database.current_objects.all()` and then use `database.name` in place of the existing variable `database_key`. Please do that everywhere that `database_key` is used in this function.

The function you're deleting, `get_non_default_database_keys`, currently gets the names of the installed databases from the `settings.py` file. The `reflect_databases()` function that's called in this function (and is defined in the same file) stores those names in the `Database` model under the `name` field. So you're getting the same values.

It seems based on your question that you aren't familiar with Django. It might be helpful to do the [Django tutorial](https://docs.djangoproject.com/en/3.2/intro/tutorial01/) to familiarize yourself with how it works, it will help you understand the Mathesar codebase better.

Reply:
thank you so much. Maybe i need to go through that tutorial.

Reply:
thanks! i will connect with you on matrix.